lib/vnet/vnet.go

```diff
@@ -176,13 +177,13 @@ type Manager struct {
 
 	// destroyed is a channel that will be closed when the VNet is in the process of being destroyed.
 	// All goroutines should terminate quickly after either this is closed or the context passed to
-	// [Manager.Run] is canceled.
+	// [NetworkStack.Run] is canceled.
 	destroyed chan struct{}
-	// wg is a [sync.WaitGroup] that keeps track of all running goroutines started by the [Manager].
+	// wg is a [sync.WaitGroup] that keeps track of all running goroutines started by the [NetworkStack].
 	wg sync.WaitGroup
 
-	// state holds all mutable state for the Manager, it is currently protect by a single RWMutex, this could
-	// be optimized as necessary.
+	// state holds all mutable state for the NetworkStack, it is currently protect by a single
+	// RWMutex, this could be optimized as necessary.
```

Comment:
nit: the mutex should be considered an implementation detail. If we change it in the future I doubt we will remember to find and update this comment.